src/ot_orchestration/operators/gce.py

```diff
@@ -0,0 +1,733 @@
+        self.zone = zone
+        self.instance_name = instance_name
+        self.labels = labels
+        self.container_image = container_image
+        self.container_command = container_command
+        self.container_args = container_args
+        self.container_env = container_env
+        self.container_service_account = container_service_account
+        self.container_scopes = container_scopes
+        self.container_files = container_files
+        self.machine_type = machine_type
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.work_disk_size_gb = work_disk_size_gb
+        self.deferrable = deferrable
+        self.poll_interval = poll_interval
+
+    def build_env_params(self):
+        """Build the environment parameters for the docker run command."""
+        return ("\n").join(
+            [f"    -e {k}={v} \\" for k, v in self.container_env.items()]
+        )
+
+    def build_volume_params(self):
+        """Build the volume parameters for the docker run command."""
+        vs = [f"    -v /home/app/{p}:{p} \\" for p in self.container_files.values()]
+        if self.work_disk_size_gb:
+            vs.append("    -v /mnt/disks/work:/mnt/disks/work \\")
+        return ("\n").join(vs)
+
+    def startup_script(self):
+        """Build the startup script for the instance.
+
+        The startup script will:
+
+        1. Create a user `app`, its home, and add it to the `docker` group.
+        2. Copy the files specified in `container_files` from GCS to the instance.
+        3. Configure docker to use the GCR credentials.
+        4. Run the containerized workload with the specified image, command, and arguments.
+        """
+        arg = (" ").join(self.container_args or [])
+
+        gcs_files = (" ").join([f'"{w}"' for w in self.container_files.keys()])
+        dest_paths = (" ").join([f'"{w}"' for w in self.container_files.values()])
+
+        init_work_disk = (
+            dedent("""
+                mkfs.ext4 -F /dev/disk/by-id/google-work-disk
+                mkdir -p /mnt/disks/work
+                mount -o discard,defaults /dev/disk/by-id/google-work-disk /mnt/disks/work
+                chmod a+w /mnt/disks/work
+            """)
+            if self.work_disk_size_gb
+            else ""
+        )
+
+        return dedent(f"""
+            #!/bin/bash
+            set -v
+            {init_work_disk}
+            useradd -m app
+            usermod -a -G docker app
+            mkdir -p /home/app
+            cd /home/app
+            orig=( {gcs_files} )
+            dest=( {dest_paths} )
+            if [ -n "${{dest[*]}}" ]; then
+                for i in "${{!dest[@]}}"; do
+                    if=${{orig[$i]}}
+                    od=$(dirname "${{dest[$i]}}")
+                    if [ "$od" = "." ]; then
+                        od=""
+                    else
+                        mkdir -p "$od"
+                    fi
```

Comment:
I think you can just get away with 
```suggestion
                   mkdir -p "$od"
```